frontend/src/pages/EditRolesPage/components/buttons/EditRolesButton.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { Profiler } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { Button } from '@SharedComponents';
+import * as ROUTES from '@Constants/routes';
+import {
+  AppUserResponse,
+  AppUserInterviewAvailability,
+  InterviewDatesResponse,
+} from '@Services/api';
+import {
+  CssBaseline,
+  Drawer,
+  AppBar,
+  Toolbar,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  Typography,
+  Divider,
+  IconButton,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogContentText,
+  DialogTitle,
+  Hidden,
+} from '@material-ui/core';
+import { updateUserById } from '@Services/UserService';
+import { AppUserPostRequest } from '@Services/api';
+
+interface EditRolesButtonProps {
+  id: number;
+  role: string;
+  email: string;
+  firstName: string;
+  lastName: string;
+}
+interface INITIAL_STATES {
+  isPopupOpen?: boolean;
+  id?: number;
+  role?: string;
+  email?: string;
+  firstName?: string;
+  lastName?: string;
+}
+
+class EditRolesButton extends React.Component<
+  EditRolesButtonProps,
+  INITIAL_STATES
+> {
+  constructor(props: EditRolesButtonProps) {
+    super(props);
+
+    this.state = {
+      isPopupOpen: false,
+      id: props.id,
+      role: props.role,
+      email: props.email,
+      firstName: props.firstName,
+      lastName: props.lastName,
+    };
+    this.handleEditRolePopup = this.handleEditRolePopup.bind(this);
+    this.handleEditRoleClose = this.handleEditRoleClose.bind(this);
+    this.updateUserRole = this.updateUserRole.bind(this);
+  }
+
+  handleEditRolePopup() {
+    this.setState({
+      isPopupOpen: true,
+    });
+  }
+  handleEditRoleClose() {
+    this.setState({
+      isPopupOpen: false,
+    });
+  }
```

Comment:
maybe sth like the below? probably don't need 2 functions
```suggestion
  setShowPopup(shouldShowPopup bool) {
    this.setState({
      isPopupOpen: shouldShowPopup,
    });
  }
```

Reply:
Would need to do some refactoring but yeah it does shorten it up a bit. I'll also implement this.